Docker networks were not properly released in E2E

## Bug Report

### Which version of ShardingSphere did you use?

master - 400e1d3ac3697561a2e4af6f4bf0dc84b682032b

### Expected behavior

Network shutdown after containers removed

### Actual behavior

E2E failed if there were many cases run in one test process.
```
org.testcontainers.containers.ContainerLaunchException: Container startup failed
Caused by: com.github.dockerjava.api.exception.NotFoundException: 
Status 404: {"message":"could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network"}
```

There were many networks without containers.

```
➜  profiling docker network ls | grep bridge
f2728444524f   1c4312e9-fbbd-435b-b6f1-f037681656f9   bridge    local
69264c8733f1   04bf11dc-c84f-4711-8b61-3fe748b4715e   bridge    local
6541bc4e262f   6dfa0b46-a7a8-49fc-bb65-7c92fc863e13   bridge    local
d1527433dedb   7f039eee-24af-48df-9e8e-ba1cc13d0b6d   bridge    local
3310d007a312   9bc4b65c-09f8-48cc-8d38-2275917fa879   bridge    local
5af6bd476367   9f1625b2-ad19-4bd3-8487-f6b7280ca49d   bridge    local
216f5521bf4e   45b3585a-30eb-46bd-81c2-f6f16cb49f6e   bridge    local
ed7c1157e904   41452a16-9eb6-486a-b1b3-e8dba755d887   bridge    local
bf966c59c839   263885bb-b3ad-4331-96f3-298d6829ee7d   bridge    local
279445c8ff5b   a987bdf0-aa33-49b0-86eb-49b8a6d556e1   bridge    local
a5b44705709a   ac9dd933-81f9-43de-8435-7edac11a65ad   bridge    local
2b97e20c823b   afaed15a-1e59-4f43-bf86-0af7521888d1   bridge    local
df2235bda9e7   b990aee9-24df-4144-9021-825b4b175010   bridge    local
9c21931a3ca7   b896073b-d831-4f93-9f0a-2824e629ac4e   bridge    local
0a432fa16e00   bridge                                 bridge    local
f5bac7c64a0c   docker_gwbridge                        bridge    local
bc77b876c980   e29478b6-433d-4818-9c94-eff645f2eec8   bridge    local
535f9d291749   f1e27b3b-294c-47af-92b5-c0195aef8501   bridge    local

➜  profiling docker inspect bf966c59c839
[
    {
        "Name": "263885bb-b3ad-4331-96f3-298d6829ee7d",
        "Id": "bf966c59c8395dfcb9ab7300a4ff2cc1b7b6276a41cb4b669be77a2d6fb875a5",
        "Created": "2023-06-25T18:10:47.308117191+08:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.20.0.0/16",
                    "Gateway": "172.20.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {},
        "Options": {},
        "Labels": {
            "org.testcontainers": "true",
            "org.testcontainers.sessionId": "786bb4e1-a675-447b-8e9f-b6e533250e12"
        }
    }
]

```

--- a/test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/ITContainers.java
+++ b/test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/ITContainers.java
@@ -110,5 +110,6 @@ public final class ITContainers implements Startable {
     public void stop() {
         embeddedContainers.forEach(Startable::close);
         dockerContainers.forEach(Startable::close);
+        network.close();
     }
 }

